pkg/dataplane/util/ansible_execution.go

```diff
@@ -142,18 +142,12 @@ func AnsibleExecution(
 
 		for _, sshKeyNodeName := range sshKeys {
 			sshKeySecret := sshKeySecrets[sshKeyNodeName]
-			if service.Spec.DeployOnAllNodeSets {
-				sshKeyName = fmt.Sprintf("ssh-key-%s", sshKeyNodeName)
-				sshKeyMountSubPath = fmt.Sprintf("ssh_key_%s", sshKeyNodeName)
-				sshKeyMountPath = fmt.Sprintf("/runner/env/ssh_key/%s", sshKeyMountSubPath)
-			} else {
-				if sshKeyNodeName != nodeSet.GetName() {
-					continue
-				}
-				sshKeyName = "ssh-key"
-				sshKeyMountSubPath = "ssh_key"
-				sshKeyMountPath = "/runner/env/ssh_key"
+			if !service.Spec.DeployOnAllNodeSets && sshKeyNodeName != nodeSet.GetName() {
+				continue
 			}
+			sshKeyName = fmt.Sprintf("ssh-key-%s", sshKeyNodeName)
+			sshKeyMountSubPath = fmt.Sprintf("ssh_key_%s", sshKeyNodeName)
+			sshKeyMountPath = fmt.Sprintf("/runner/env/ssh_key/%s", sshKeyMountSubPath)
```

Comment:
we probably shouldn't be using /runner/env/ssh_key as a directory at all, as it's meant to be a file for ansible-runner:

https://ansible.readthedocs.io/projects/runner/en/stable/intro/#env-ssh-key

thankfully ansible-runner handles this gracefully if it can't load a file, and just ignores whatever is there. that's why I suggested putting these keys in a different directory on the initial PR that added this code: https://github.com/openstack-k8s-operators/dataplane-operator/pull/624#discussion_r1491635975

Regardless I don't know what issue this is fixing because if /runner/env/ssh_key *is* a file (as in the case of a non-global service), then ansible-runner automatically adds the key there to the ssh-agent. At that point, it doesn't matter that ansible_ssh_private_key in the inventory is pointing to a non-existant file. I agree that would be good to clean up so it's more obvious what key actually gets used, I'm just not sure that this fixes the issue.

Reply:
> we probably shouldn't be using /runner/env/ssh_key as a directory at all, as it's meant to be a file for ansible-runner

Agreed, but we should do that in a separate PR as we hardcode the path in kuttl asserts and would need changes to all kuttl tests and this is trying for to fix a specific issue reported downstream.

>  At that point, it doesn't matter that ansible_ssh_private_key in the inventory is pointing to a non-existant file.

Probably it matters as reported in downstream testing in https://redhat-internal.slack.com/archives/C063J9ZJ7N3/p1719482555872529 (Note: I've not checked the ansible-runner/ansible combination causing the issue), but I think this PR would fix the issue if encountered. I've added a jira too.

Reply:
The error there:

`
"msg": "Failed to connect to the host via ssh: Warning: Permanently added '192.168.122.100' (ED25519) to the list of known hosts.\r\nno such identity: /runner/env/ssh_key/ssh_key_openstack-edpm-ipam: Not a directory\r\ncloud-admin@192.168.122.100: Permission denied (publickey,gssapi-keyex,gssapi-with-mic,password)."
`

is because `/runner/env/ssh_key` is not correct, so it falls back to try the path defined in the inventory. The cause sounds like ci-framework keeps recreating the ssh key secret, and it no longer contains the key configured in authorized_keys on the edpm node. A way to confirm this is to check those keys in the secret and in authorized_keys and see if they match, if they don't, that's the problem.

Reply:
Did you see the workaround mentioned in https://issues.redhat.com/browse/OSPRH-8057?focusedId=25012795&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-25012795. That sounds like the secret in `/runner/env/ssh_key` is good, though it's good to validate if the key is good or not.

Reply:
Yes, I did see that. The workaround doesn't make sense and I question if it actually works. Editing the inventory isn't going to persist, the inventory will just be regenerated on the next NodeSet reconcile which is going to keep happening during a Deployment.

the comment also says "Hopefully on the next restart...", I doubt it actually does.

Reply:
What's the reasoning behind the fix is why I'm questioning this. 

The PR implies that  `/runner/env/ssh_key` can be correct, but if  `ansible_ssh_private_key_file` points to a non-existent path, then the Deployment doesn't work. If that was the case then no Deployment would ever work as things stand.

Reply:
> The PR implies that `/runner/env/ssh_key` can be correct, but if `ansible_ssh_private_key_file` points to a non-existent path, then the Deployment doesn't work.  If that was the case then no Deployment would ever work as things stand.

Well, my initial suspicion as mentioned in the slack thread was that it could be some recent change in ansible-runner/ansible on why it's not working now, but I did not dig into it. This PR may not fix the issue reported, but it would ensure that we don't get misleading error like `no such identity: /runner/env/ssh_key/ssh_key_openstack-edpm-ipam: Not a directory` when the ssh key is not valid. That would at least be fixed with this PR as the end user won't be aware of the fallback and 'ansible_ssh_private_key_file' pointing to an non-existent location.